File: Leetcode/2439 Minimize Maximum of Array.ts
function minimizeArrayValue(nums: number[]): number {
    let prefixSum = nums[0], maxValue = nums[0];
    for(let i = 1; i < nums.length; i++) {
        prefixSum += nums[i];
        const value = Math.ceil(prefixSum / (i + 1));
        maxValue = Math.max(maxValue, value);
    }
    return maxValue;
}

File: Leetcode/2439 Minimize Maximum of Array.py
class Solution:
    def minimizeArrayValue(self, nums):
        prefix_sum, max_value = 0, 0
        for i, num in enumerate(nums):
            prefix_sum += num
            value = ceil(prefix_sum / (i + 1))
            max_value = max(max_value, value)
        return max_value
